src/mono/mono/mini/mini-arm64.h

```diff
@@ -23,6 +23,8 @@
 
 #if !defined(DISABLE_SIMD)
 #define MONO_ARCH_SIMD_INTRINSICS 1
+#define MONO_ARCH_NEED_SIMD_BANK 1
+#define MONO_ARCH_USE_SHARED_FP_SIMD_BANK 1
 #endif
```

Comment:
This is not going to work this way, because on arm64, 2 fp regs overlap 1 simd reg, i.e. q0 overlaps d0/d1, q1 overlaps d2/d3, etc.

Reply:
I believe you're thinking of Arm32. Arm64 does not overlap in this way.

----------------

That is, Arm32 has `32x 64-bit doubleword registers: D0-D31`. To allocate a 128-bit register `Q0` you must allocate 2x 64-bit registers `D0 & D1` and so therefore there are `16x 128-bit quadword registers: Q0-Q15`

The `32x 64-bit doubleword registers` can then be used as `32x 64-bit doubleword registers: D0-D31` or as `32x 32-bit singleword registers: S0-S31`

----------------

However, on Arm64 there is simply `32x 128-bit quadword registers: V0-V31`. These can be used as `32x 64-bit doubleword registers` or as `32x 32-bit singleword registers`, but they all represent the same register allocation (much as XMM/YMM/ZMM and the "scalar floating-point registers" operate for xarch).

-------------------

The architecture manuals go into this in more detail and have diagrams showing the differences.